processor.go

```diff
@@ -45,33 +45,36 @@ func NewProcessor(net Network, graph Graph, build Builder, strat Strategy, sign
 		sign:   sign,
 		verify: verify,
 		cache:  cache,
-	}
+	} //  processor instantiation; fields can be set to anything that implements the corresponding interfaces
 
 	return &pro
 }
 
 func (pro *Processor) Bootstrap() error {
 
 	// get current tip of the state
-	tip, err := pro.graph.Tip()
+	tip, err := pro.graph.Tip() // obtain the tip of the DAG upon which to build
+	// in this function we want to bootstrap from genesis
 	if err != nil {
 		return rich.Errorf("could not get tip: %w", err)
 	}
 
 	// check the tip is at zero
 	if tip.Height != 0 {
+		// check that the received tip is the first vertex/genesis vertex with zero heigth
 		return rich.Errorf("invalid tip height").Uint64("tip_height", tip.Height)
 	}
 
 	// vote on the vertex
-	err = pro.castVote(tip)
+	err = pro.castVote(tip) // cast own vote
```

Comment:
Yes, but the point here is that, when bootstrapping, the genesis block is not yet signed by any of the validators. They sign it explicitly as part of the start-up logic. If anyone's genesis block differs, they don't agree and bootstrapping of the network fails.